### PR TITLE
examples: button with tooltip

### PR DESCRIPTION
**Pull request by frassinier**
Add an example using a Button with Tooltip and VisuallyHidden.

button-with-tooltip

It will increase the list of examples related to #626

### CODE REVIEW

File: packages/reakit/src/Button/__examples__/ButtonWithTootltip/__tests__/index-test.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  expect(buttonUnlocked).toBeVisible();
+  press.Tab();
+  expect(text("Click to lock")).toBeVisible();
+  expect(text("It's unlocked!")).toBeVisible();
+  click(buttonUnlocked);
+  const buttonLocked = text("🔒");
+  expect(buttonLocked).toBeVisible();
+  expect(text("Click to unlock")).toBeVisible();
+  expect(text("It's locked!")).toBeVisible();
+  press.Space(buttonLocked);
```
Comment (diegohaz) [inline packages/reakit/src/Button/__examples__/ButtonWithTootltip/__tests__/index-test.tsx]: If it already has focus, you don't need to pass the element.
```suggestion
  press.Space();
```

File: packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import { Button } from "reakit/Button";
+import { Tooltip, TooltipReference, useTooltipState } from "reakit/Tooltip";
+import { VisuallyHidden } from "reakit/VisuallyHidden";
+
+type Props = {
+  isLocked: boolean;
+  onClick: () => void;
+};
```
Comment (diegohaz) [inline packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]:
```suggestion
import {
  Tooltip,
  TooltipReference,
  useTooltipState,
  TooltipReferenceHTMLProps,
} from "reakit/Tooltip";
import { VisuallyHidden } from "reakit/VisuallyHidden";

type Props = TooltipReferenceHTMLProps & {
  isLocked?: boolean;
};
```

File: packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+function LockButton({ isLocked, ...rest }: Props) {
+  const tooltip = useTooltipState();
+  return (
+    <>
+      <TooltipReference {...rest} {...tooltip} as={Button}>
```
Comment (diegohaz) [inline packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]: ~~Let's push the pattern of always putting the state object first so people can override state props if they want.~~ Let's just rename it to `props`.
```suggestion
function LockButton({ isLocked, ...props }: Props) {
  const tooltip = useTooltipState();
  return (
    <>
      <TooltipReference {...props} {...tooltip} as={Button}>
```

File: packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          <>
+            <VisuallyHidden>Click to unlock</VisuallyHidden>{" "}
+            <span aria-hidden>🔒</span>
+          </>
+        ) : (
+          <>
+            <VisuallyHidden>Click to lock</VisuallyHidden>{" "}
+            <span aria-hidden>🔓</span>
+          </>
+        )}
```
Comment (diegohaz) [inline packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]: Nice usage of `VisuallyHidden` and `<span aria-hidden>` 👍

Could we make it more simple? Maybe something like this (haven't tested it)?
```suggestion
        <>
          <VisuallyHidden>Click to {isLocked ? "unlock" : "lock"}</VisuallyHidden>{" "}
          <span aria-hidden>{isLocked ? "🔒" : "🔓"}</span>
        </>
```

We can also abstract these labels into variables outside the JSX.

File: packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      </TooltipReference>
+      <Tooltip {...tooltip}>{`It's ${
+        isLocked ? "locked" : "unlocked"
+      }!`}</Tooltip>
```
Comment (diegohaz) [inline packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]:
```suggestion
      <Tooltip {...tooltip}>
        It's {isLocked ? "locked" : "unlocked"}!
      </Tooltip>
```
Or abstracting the label into a variable outside of the JSX.

Comment (frassinier) [reply packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]: I will use your suggestion 👌

File: packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx (hunk 3 above)
Comment (diegohaz) [reply packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]: Actually, nevermind! In this case I think we should indeed put `{...props}` first as we're only accepting HTML props. Updated the suggestion.

Comment (frassinier) [reply packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx]: I was not decided about it too :)
But, yes, I rename it.

**Comment by frassinier** [comment]
Thank you @Haz for taking the time for the review! 🤗

**Comment by haz** [comment]
I think you have the wrong haz :p

**Comment by frassinier** [comment]
I'm stuck using "It's" by ESLint which prefers escaped apostrophe.
 
>ESLint: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.(react/no-unescaped-entities)

Which fix do you recommend? I would suggest keeping the template string format?

**Comment by diegohaz** [comment]
I would do something like this:
```jsx
...
const tip = `It's ${isLocked ? "locked" : "unlocked"}!`;
return (
  ...
  <Tooltip>{tip}</Tooltip>
);
```

**Pull request merged**